app/main/views.py

```diff
@@ -146,13 +249,13 @@ def pagination_links(pagination, endpoint, args):
     ]
 
 
-def get_json_from_request():
-    if request.content_type != 'application/json':
+def get_json_from_request(key):
```

Comment:
Looked at this - split it up so there are two methods (not completly happy and think there is a JSONParse template solution - but I'm pushed for time prior to holidays).

The existing method does basic checks, and the new method checks for keys.